data/data/powervs/main.tf

```diff
@@ -42,5 +44,18 @@ module "loadbalancer" {
 
   # TODO add resources for master/controller
   master_ips = []
+}
+
+
+module "dns" {
+  source = "./dns"
 
+  cluster_id                 = var.powervs_cluster_id
+  cluster_domain             = var.powervs_cluster_domain
```

Comment:
Lets make use of base_domain and cluster_domain from the parent directory.

Reply:
@mkumatag I see `base_domain` being used only for aws and azure to specifically create public hosted zone. I don't see that distinction for powervs. 
Do you still think we need to split into base_domain and cluster_domain ?

Reply:
these values are used in the installer code heavily, we need to ensure the right mapping for the entried in the DNS, e.g:

api-init is mapped as follows:

https://github.com/openshift/installer/blob/6d778f911e79afad8ba2ff4301eda5b5cf4d8e9e/pkg/asset/ignition/machine/node.go#L30

```golang
ignitionHost = fmt.Sprintf("api-int.%s:22623", installConfig.ClusterDomain())
```

and clusterdomain is nothing but a <clustername>.<basename>

```golang
// ClusterDomain returns the DNS domain that all records for a cluster must belong to.
func (c *InstallConfig) ClusterDomain() string {
	return fmt.Sprintf("%s.%s", c.ObjectMeta.Name, strings.TrimSuffix(c.BaseDomain, "."))
}
```

hence atleast we need to ensure we follow these convention to align with the installer code.

Reply:
The installer code should handle this imho.